web/src/pages/BookWizard/GeneralPage.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useState, useEffect, useRef } from 'react';
+import '../../App.css';
+import styles from './GeneralPage.module.css';
+import wizardStyles from '../BookWizardPage.module.css';
+import requiredFieldImage from '../../assets/images/star-of-life-solid.svg';
+
+type GeneralPageProps = {
+  onTitleChange: ( data: string ) => void
+  onAuthorChange: (data: string) => void 
+  onImageChange: (data: File | null) => void
+  // the current state of the fields. Used because fields dissapear when component re-renders
+  title: string 
+  author: string
+  image: File | null 
+};
+
+/**
+ * General page for the book wizard. Inputs for title, author, and image 
+ */
+export const GeneralPage: React.FC<GeneralPageProps> = ({onTitleChange, onAuthorChange, onImageChange, title, author, image}) => {
+  const [previewUrl, setPreviewUrl] = useState<string>("");
+  const fileInput = useRef<HTMLInputElement>(null);
+  const acceptableMimeType = ["image/png", "image/jpeg"];
+
+  useEffect( () => {
+    onTitleChange(title);
+  }, [title]);
+
+  useEffect( () => {
+    onAuthorChange(author);
+  }, [author]);
+
+  useEffect( () => {
+    onImageChange(image);
+  }, [image]);
+
+  useEffect( () => {
+    if (image != null) {
+      setPreviewUrl(URL.createObjectURL(image));
+    }
+  }, []);
+
+
+  const handleOnDragOver = (e: React.DragEvent): void => {
+    e.preventDefault();
+  };
+
+  const handleOnDrop = (e: React.DragEvent): void => {
+    e.preventDefault();
+    e.stopPropagation();
+    const imageFile = e.dataTransfer.files[0];
+    if(acceptableMimeType.indexOf(imageFile.type) == -1) {
+      alert("Please enter a jpeg or a png file");
+      return;
+    } 
+    else {
+      handleFile(imageFile);
+    }
+  };
+
+  const handleFile = (file: File): void => {
```

Comment:
Uploading an image, then clicking the image to upload again, then clicking cancel gives this error:
> TypeError: Failed to execute 'createObjectURL' on 'URL': Overload resolution failed.

Could a fix just be to add a null check? :
```
    if (file != null){ 
      onImageChange(file);   
      setPreviewUrl(URL.createObjectURL(file));   
    }  
```